common/interfaces/card_interface/nfc_events.c

```diff
@@ -0,0 +1,154 @@
+ *
+ * For purposes of the foregoing, "Sell" means practicing any or all
+ * of the rights granted to you under the License to provide to third
+ * parties, for a fee or other consideration (including without
+ * limitation fees for hosting or consulting/ support services related
+ * to the Software), a product or service whose value derives, entirely
+ * or substantially, from the functionality of the Software. Any license
+ * notice or attribution required by the License must also include
+ * this Commons Clause License Condition notice.
+ *
+ * Software: All X1Wallet associated files.
+ * License: MIT
+ * Licensor: HODL TECH PTE LTD
+ *
+ ******************************************************************************
+ */
+
+/*****************************************************************************
+ * INCLUDES
+ *****************************************************************************/
+#include "nfc_events.h"
+
+#include "memzero.h"
+#include "string.h"
+/*****************************************************************************
+ * EXTERN VARIABLES
+ *****************************************************************************/
+
+/*****************************************************************************
+ * PRIVATE MACROS AND DEFINES
+ *****************************************************************************/
+#define DEFAULT_NFC_TIMEOUT 100
+
+/*****************************************************************************
+ * PRIVATE TYPEDEFS
+ *****************************************************************************/
+
+/*****************************************************************************
+ * STATIC VARIABLES
+ *****************************************************************************/
+static nfc_ctx_t nfc_ctx;
+static nfc_event_t nfc_event;
+
+/*****************************************************************************
+ * GLOBAL VARIABLES
+ *****************************************************************************/
+
+/*****************************************************************************
+ * STATIC FUNCTION PROTOTYPES
+ *****************************************************************************/
+
+/*****************************************************************************
+ * STATIC FUNCTIONS
+ *****************************************************************************/
+
+/*****************************************************************************
+ * GLOBAL FUNCTIONS
+ *****************************************************************************/
+bool nfc_get_event(nfc_event_t *nfc_event_os_obj) {
+  ASSERT(nfc_event_os_obj != NULL);
+
+  if (nfc_event.event_occured == true) {
+    memcpy(nfc_event_os_obj, &nfc_event, sizeof(nfc_event_t));
+    nfc_reset_event();
+    return true;
+  }
+  return false;
+}
+
+void nfc_reset_event() {
+  memzero(&nfc_event, sizeof(nfc_event_t));
+}
+
+void nfc_enable_card_detect_event() {
+  nfc_ctx.card_detect_enabled = true;
```

Comment:
Please reconsider if flag `card_detect_enabled` is still required or not

Reply:
Agreed, I'll do the cleanup on this.

Reply:
Agreed, will remove.